packages/editor/src/core/extensions/custom-image/components/image-uploader.tsx

```diff
@@ -44,42 +119,38 @@ export const CustomImageUploader = (props: {
     [uploadFile]
   );
 
-  useEffect(() => {
-    // no need to validate as the file is already validated before the drop onto
-    // the editor
-    if (existingFile) {
-      uploadFile(existingFile);
-    }
-  }, [existingFile, uploadFile]);
-
   return (
     <div
       className={cn(
         "image-upload-component flex items-center justify-start gap-2 py-3 px-2 rounded-lg text-custom-text-300 hover:text-custom-text-200 bg-custom-background-90 hover:bg-custom-background-80 border border-dashed border-custom-border-300 cursor-pointer transition-all duration-200 ease-in-out",
         {
           "bg-custom-background-80 text-custom-text-200": draggedInside,
-        },
-        {
           "text-custom-primary-200 bg-custom-primary-100/10": selected,
+          "text-red-500 bg-red-500/5 hover:bg-red-500/5 hover:text-red-500 border-red-400 border-dashed":
+            failedToLoadImage,
         }
       )}
       onDrop={onDrop}
       onDragOver={onDragEnter}
       onDragLeave={onDragLeave}
       contentEditable={false}
-      onClick={handleUploadClick}
+      onClick={() => fileInputRef.current?.click()}
     >
       <ImageIcon className="size-4" />
       <div className="text-base font-medium">
-        {loading ? "Uploading..." : draggedInside ? "Drop image here" : existingFile ? "Uploading..." : "Add an image"}
+        {failedToLoadImage
```

Comment:
please avoid this nested ternary logic. simplify it with if else and comments. Also move it out to a different function if possible

Reply:
Sure thing, refactored this onto a function.